x-pack/qa/sql/src/main/java/org/elasticsearch/xpack/qa/sql/jdbc/ResultSetTestCase.java

```diff
@@ -6,16 +6,1082 @@
 package org.elasticsearch.xpack.qa.sql.jdbc;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.sql.Blob;
+import java.sql.Clob;
 import java.sql.Connection;
+import java.sql.NClob;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
 import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.Calendar;
 import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
 
-public class ResultSetTestCase extends JdbcIntegrationTestCase {
-    public void testGettingTimestamp() throws Exception {
+import static java.lang.String.format;
+import static java.util.Calendar.DAY_OF_MONTH;
+import static java.util.Calendar.ERA;
+import static java.util.Calendar.HOUR_OF_DAY;
+import static java.util.Calendar.MILLISECOND;
+import static java.util.Calendar.MINUTE;
+import static java.util.Calendar.MONTH;
+import static java.util.Calendar.SECOND;
+import static java.util.Calendar.YEAR;
+
+public class ResultSetTestCase extends ResultSetBaseTestCase {
```

Comment:
There is quite a lot of repetition in this class. I wonder if there is a way to create some helper methods that take functions/lambdas to reduce it? There is a balance between reducing repetition and making the test easy to follow but I wonder if there are even some smaller things we can do to help here?